src/Microsoft.AspNet.Cors.Core/RequestExtensions.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNet.Http;
+
+namespace Microsoft.AspNet.Cors.Core
+{
+    /// <summary>
+    /// Extension methods for <see cref="HttpRequest"/>.
+    /// </summary>
+    public static class RequestExtensions
+    {
+        /// <summary>
+        /// Determines if the current <see cref="HttpRequest"/> is a pre-flight request.
+        /// </summary>
+        /// <param name="request">The given <see cref="HttpRequest"/>.</param>
+        /// <returns><c>true</c> if there is an Origin header and there is a non null Access-Control-Request-Method
+        /// header and the request method is OPTIONS.<c>false</c> otherwise.</returns>
+        public static bool IsPreflight(this HttpRequest request)
```

Comment:
Don't use extension methods for this, just look for method == "OPTIONS" at the appropriate place.